### PR TITLE
openfga-cli: init at 0.4.0

### PR DESCRIPTION
**Pull request by jlesquembre**
## Description of changes

Add openfga-cli
https://github.com/openfga/cli
https://openfga.dev/

Related to #310534

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openfga-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+}:
```
Comment (DontEatOreo) [inline pkgs/by-name/op/openfga-cli/package.nix:5]:
```suggestion
{
  lib,
  buildGoModule,
  fetchFromGitHub,
  installShellFiles,
}:
```
Formatted as per RFC 0166. You can use the package `nixfmt-rfc-style` to automatically comply with the RFC in the future

File: pkgs/by-name/op/openfga-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+buildGoModule rec {
+  pname = "openfga-cli";
+  version = "0.4.0";
```
Comment (DontEatOreo) [inline pkgs/by-name/op/openfga-cli/package.nix:9]:
```suggestion
let
  pname = "openfga-cli";
  version = "0.4.0";
in
buildGoModule {
```

Using `rec` is discouraged, you can learn more about it here: https://nix.dev/guides/best-practices#recursive-attribute-set-rec

File: pkgs/by-name/op/openfga-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    $out/bin/fga completion zsh > $completions_dir/_fga.zsh
+    $out/bin/fga completion fish > $completions_dir/fga.fish
+    installShellCompletion $completions_dir/*
+  '';
+
+  meta = with lib; {
+    description = "A cross-platform CLI to interact with an OpenFGA server";
+    license = licenses.asl20;
+    homepage = "https://github.com/openfga/cli";
+    maintainers = with maintainers; [ jlesquembre ];
```
Comment (DontEatOreo) [inline pkgs/by-name/op/openfga-cli/package.nix:47]:
```suggestion
  meta = {
    description = "A cross-platform CLI to interact with an OpenFGA server";
    homepage = "https://github.com/openfga/cli";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ jlesquembre ];
```
Using `with` is discouraged as well, you can read more about it here: https://nix.dev/guides/best-practices#with-scopes and #293767

**Review by 4evy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/op/openfga-cli/package.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/by-name/op/openfga-cli/package.nix]: Thanks, I missed that one, I was still using nixpkgs-fmt

File: pkgs/by-name/op/openfga-cli/package.nix (hunk 3 above)
Comment (jlesquembre) [reply pkgs/by-name/op/openfga-cli/package.nix]: Done

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 310184` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>openfga-cli</li>
  </ul>
</details>

**Review by 4evy — APPROVED** [review]

**Review by marcusramberg — APPROVED** [review]
lgtm. works on darwin as well.

```
1 package built:
openfga-cli
```

**Pull request merged**